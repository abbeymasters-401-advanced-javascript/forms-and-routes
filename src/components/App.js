import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Header from './Header';
import Footer from './Footer';
import Home from '../containers/Home';
import List from './List';
import Detail from './Detail';

export default function App() {
  return (
    <>
      <Router>
        <>
          <Header />
          <Switch>
            <Route exact path='/' component={Home}></Route>
            <Route path='list/:character' component={List}></Route>
            <Route exact path='/:characterName' component={Detail}></Route>
          </Switch>
          <Footer />
        </>
      </Router>
    </>
  );
}
