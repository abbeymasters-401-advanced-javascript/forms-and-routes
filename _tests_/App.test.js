import React from 'react';
import { shallow } from 'enzyme';
import App from '../src/components/App';

describe('App Component', () => {
  it('renders App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
