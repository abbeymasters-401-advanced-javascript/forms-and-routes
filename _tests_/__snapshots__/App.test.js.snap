// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders App 1`] = `
<Fragment>
  <BrowserRouter>
    <Header />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/character/:id"
      />
      <Route
        component={[Function]}
        path="/list/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
    </Switch>
    <Footer />
  </BrowserRouter>
</Fragment>
`;
